homeassistant/components/sensor/isy994.py

```diff
@@ -0,0 +1,90 @@
+""" Support for ISY994 sensors. """
+# system imports
+import logging
+
+# homeassistant imports
+from homeassistant.components.isy994 import (ISY, ISYDeviceABC, SENSOR_STRING,
+                                             HIDDEN_STRING)
+from homeassistant.const import (STATE_OPEN, STATE_CLOSED, STATE_HOME,
+                                 STATE_NOT_HOME, STATE_ON, STATE_OFF)
+
+DEFAULT_HIDDEN_WEATHER = ['Temperature_High', 'Temperature_Low', 'Feels_Like',
+                          'Temperature_Average', 'Pressure', 'Dew_Point',
+                          'Gust_Speed', 'Evapotranspiration',
+                          'Irrigation_Requirement', 'Water_Deficit_Yesterday',
+                          'Elevation', 'Average_Temperature_Tomorrow',
+                          'High_Temperature_Tomorrow',
+                          'Low_Temperature_Tomorrow', 'Humidity_Tomorrow',
+                          'Wind_Speed_Tomorrow', 'Gust_Speed_Tomorrow',
+                          'Rain_Tomorrow', 'Snow_Tomorrow',
+                          'Forecast_Average_Temperature',
+                          'Forecast_High_Temperature',
+                          'Forecast_Low_Temperature', 'Forecast_Humidity',
+                          'Forecast_Rain', 'Forecast_Snow']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """ Sets up the isy994 platform. """
+    # pylint: disable=protected-access
+    logger = logging.getLogger(__name__)
+    devs = []
+    # verify connection
+    if ISY is None or not ISY.connected:
+        logger.error('A connection has not been made to the ISY controller.')
+        return False
+
+    # import weather
+    if ISY.climate is not None:
+        for prop in ISY.climate._id2name:
+            if prop is not None:
+                prefix = HIDDEN_STRING \
+                    if prop in DEFAULT_HIDDEN_WEATHER else ''
+                node = WeatherPseudoNode('ISY.weather.' + prop, prefix + prop,
+                                         getattr(ISY.climate, prop),
+                                         getattr(ISY.climate, prop + '_units'))
+                devs.append(ISYSensorDevice(node))
+
+    # import sensor nodes
+    for (path, node) in ISY.nodes:
+        if SENSOR_STRING in node.name:
+            if HIDDEN_STRING in path:
+                node.name += HIDDEN_STRING
+            devs.append(ISYSensorDevice(node, [STATE_ON, STATE_OFF]))
+
+    # import sensor programs
+    for (folder_name, states) in (
```

Comment:
Does this require users to have setup their ISY in a specific way? We should make sure to document this properly.